server/sandbox_status_test.go

```diff
@@ -0,0 +1,105 @@
+package server
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/kubernetes-incubator/cri-o/lib"
+	"github.com/kubernetes-incubator/cri-o/lib/sandbox"
+	"github.com/kubernetes-incubator/cri-o/oci"
+	"github.com/kubernetes-incubator/cri-o/version"
+	pb "k8s.io/kubernetes/pkg/kubelet/apis/cri/runtime/v1alpha2"
+)
+
+const testID = "test-id"
```

Comment:
Hard-coding the test ID seems like it might lead to conflicts (e.g. with `go test -parallel …`).  Can we either get per-test names (e.g. `server-sandbox-status-1`, `server-sandbox-status-2`, …) or [use UUIDs like runtime-tools][1]?

[1]: https://github.com/opencontainers/runtime-tools/blob/v0.5.0/validation/util/test.go#L182

Reply:
It's just a test with two subtests and the container and sandbox are just created once, I don't think we will have that issue.

However, for the sake of reusability I've changed `newTestSandboxOrFailNow` and `newTestContainerOrFailNow` to receive an ID instead using the const, that way we could easily use it for other tests in the same package or move it to some fixtures place.

Are you ok with the change or do you still want the random part added? I could do that as well.

In the meantime, I am pushing the changes to these functions.

Thanks for the review!

Reply:
🤔I was thinking about this and what do you think about generating the ID in `newTestSandboxOrFailNow` and `newTestContainerOrFailNow` and return it? Something in the form:

```
func newTestSandboxOrFailNow(t *testing.T) (id, *sandbox.Sandbox) {
	id := // put something random here
	sb, err := sandbox.New(id, ...)
	if err != nil {
		t.Fatal(err)
	}
	return id, sb
}

func newTestContainerOrFailNow(t *testing.T) *oci.Container {
	c, err := oci.NewContainer(id, ...)
	if err != nil {
		t.Fatal(err)
	}
	return id, c
}
```